libyul/optimiser/LoopInvariantCodeMotion.h

```diff
@@ -56,7 +56,11 @@ class LoopInvariantCodeMotion: public ASTModifier
 	{ }
 
 	/// @returns true if the given variable declaration can be moved to in front of the loop.
-	bool canBePromoted(VariableDeclaration const& _varDecl, std::set<YulString> const& _varsDefinedInCurrentScope) const;
+	bool canBePromoted(
+		VariableDeclaration const& _varDecl,
+		std::set<YulString> const& _varsDefinedInCurrentScope,
+		SideEffectsCollector const& _blockSideEffects
```

Comment:
Can you explain this parameter in a comment? Why is it a `SideEffectsCollector` and not just a `SideEffect`?

Reply:
`SideEffect` can also be used. I don't have strong reasons for or against this choice. I'll have to write a getter for `m_sideEffects` in `SideEffectCollector` if we change it to `SideEffect`.

Reply:
That's probably a good idea. I think `SideEffectsCollector` is older that the `SideEffects` struct - that's why there is no getter. Maybe we should refactor it to just use the struct instead of the functions.